pyccel/parser/semantic.py

```diff
@@ -1869,7 +1867,15 @@ def _visit_Assign(self, expr, **settings):
 
         for l, r in zip(lhs,rhs):
             is_pointer_i = l.is_pointer if isinstance(l, (Variable, DottedVariable)) else is_pointer
-
+            if isinstance(l, Variable) and l.is_ndarray:
```

Comment:
`l` is always a `Variable` I think

Reply:
i think it can be a DottedVariable from what I saw from the previous line

Reply:
Ah you are probably right. But a `DottedVariable` (should) have the same properties as a Variable and it can be an ndarray too. Imagine an array as a member of a class

Reply:
Does this mean `DottedVariable` is missing a property?

Reply:
@saniQQ ?

Reply:
if it should have the same properties then yeah it's missing some of the Variable Class properties, I am not so sure to when we use the DottedVariable that's why I can't know for sure if we need the nd_array property

Reply:
We use it when there is a dot in the python expression (so `a.b`). As in C/C++ this happens when `a` is a class instance, but in python it can also happen when `a` is a module (e.g. numpy). It is made of a lhs (`a` in our example) and a rhs (`b` in our example). `b` is a variable so really a `DottedVariable` should have all the same properties. In reality that is hard to maintain so I think we often try to use `rhs` to get the `Variable` properties.

So maybe the line above should be:
```
if (isinstance(l, Variable) and l.is_ndarray) or \
   (isinstance(l, DottedVariable) and l.rhs.is_ndarray):
```

but this isn't very important at the moment given that classes aren't implemented yet

Reply:
i will  keep that in mind for next time as that piece of code now is removed